<?xml version="1.0"?>
<launch>

  <node pkg="mipp_packages" type="ugv_planner_node" name="rrt_planner_node" output="screen">
    <!-- Parameters -->
    <param name="planner_world_frame"       value="map"/>
    <param name="goal_sample_probability"   value="0.2"/>
    <param name="goal_radius"               value="0.01"/>
    <param name="ugv_midpoint_z"            value="0.35"/>
    <param name="x_range_min"               value="-10.0"/>
    <param name="x_range_max"               value="10.0"/>
    <param name="y_range_min"               value="-10.0"/>
    <param name="y_range_max"               value="10.0"/>
    <param name="z_range_min"               value="0.35"/>
    <param name="z_range_max"               value="0.35"/>
    <param name="planner_rate"              value="100.0"/>
    <param name="planner_algorithm"         value="1"/> <!-- 1:RRT, 2:RRTstar, 3:Informed RRTstar -->
    <param name="planner_max_tree_nodes"    value="2000"/> 
    <param name="planner_max_time"          value="10.0"/> 
    <param name="max_ray_distance"          value="2.5"/>

    <!-- Remap topics -->
    <!-- Published topics -->
    <remap from="rrt_planner/random_point" to="rrt_planner/random_point"/>
  </node>

  <node pkg="octomap_server" type="octomap_server_node" name="octomap_server_node" output="screen" args="$(find mipp_packages)/rviz/playpen_octomap.bt">
    <!-- Parameters -->
  </node>

  <!--- 
    Run Rviz 
  -->
  <node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find mipp_packages)/rviz/rrt_planner.rviz"/>

</launch>

